ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -751,7 +751,7 @@ static char *make_filename_safe(const char *filename)
 		}
 		return estrdup(filename);
 	}
-	if (*filename && memcmp(filename, ":memory:", sizeof(":memory:"))) {
+	if (*filename && (sizeof(filename) != sizeof(":memory:") || memcmp(filename, ":memory:", sizeof(":memory:")) != 0)) {
```

Comment:
`))))` was too hard to see, so I added `!= 0`. The meaning remains the same.

Reply:
This is wrong, `sizeof(filename)` is not the length, it is the size of the pointer.
I also think the `!= 0` addition is wrong for `memcmp`?

Reply:
why bothering with memcmp anyway ? why not just any str*cmp ?

Reply:
@nielsdos 
> This is wrong, sizeof(filename) is not the length, it is the size of the pointer.

Right. I entered sizeof with the intention of strlen...

> I also think the != 0 addition is wrong for memcmp?

Since it becomes true when `1` and `-1`, it seems to be the same because it becomes true when `!= 0`, but am I wrong?

@devnexen 

>why bothering with memcmp anyway ? why not just any str*cmp ?

I thought I had to convert it to zend_string, but there were some that could be used as char, so I fixed it.

Reply:
Right, I misread the original condition. So !=0 is fine.